android/src/org/ppsspp/ppsspp/PpssppActivity.java

```diff
@@ -112,4 +113,16 @@ public void run() {
 			}
 		});
 	}
+
+	public int openContentUri(String uriString) {
+		try {
+			Uri uri = Uri.parse(uriString);
+			ParcelFileDescriptor filePfd = getContentResolver().openFileDescriptor(uri, "r");
+			int fd = filePfd.detachFd();  // Take ownership of the fd.
```

Comment:
Hm, I think this'd throw if `filePfd` is null, but probably better to check?

-[Unknown]

Reply:
The catch should take care of it, but yeah, probably worth checking directly.